### PR TITLE
Rewrite error handle with pkg/errors

### PR DESCRIPTION
**Pull request by babygoat**
Summary the changes

- [x] Remove unused files (prefix with _ files)
- [x] Remove deprecated routes (/v1/auth) and rewrite test with v1 token
- [x] Define errors usage guide
- [x] Define log usage guide

errors usage
* errors.New when creating an error object with stack trace information
* errors.Wrap when wrapping error with standard/3rd party error with customized context message and stack trace information(e.g, gin binding returns error)
* errors.WithStack when wrapping error with stacktrace only. No need to annotate extra context message
* errors.WithMessage when wrapping error with context message but no additional message

In most cases, we only need errors.New/errors.Wrap/errors.WithStack in the very first place.(e.g., if db operation reports fail, the storage layer is the first place to handle). Then, any intermediate layer just simply returns err to the upper layer.(i.e., avoid using errors.WithMessage which do not gain any benefit during error handling). In the topmost place, we can printing the error with format string “%+v” (with stack) “%v”(context messages only) “%q” quoted context message

log severity usage
[Level]

ERROR
INFO
DEBUG

[Usage]

Target User: developer, other members in organization
Object: For highlighting the routine information like application progress
      status. Also, for information required for customer service(e.g.,
      activated token expired)

Target User: developer
Object: More in-depeth information for further investigation but frequently
        printing will be noisy. (e.g., ORM translated sql statements)

Target User: developer
Object: Quickly identifying the issue emerge from the stacktrace info and
        details.

In developement, application logs will print in all levels but suppress the debug
information in production.

### CODE REVIEW

**Comment by babygoat** [comment]
Rebased and updated.

File: storage/bookmark.go [hunk 1]
```diff
@@ -16,126 +18,103 @@ func (g *GormStorage) GetABookmarkBySlug(slug string) (models.Bookmark, error) {
...
 	}
 
 	// update the bookmark fields
-	err = g.db.Model(&_bookmark).Updates(bookmark).Error
+	/*err = g.db.Model(&_bookmark).Updates(bookmark).Error
 
 	if err != nil {
 		return _bookmark, g.NewStorageError(err, "GormStorage.CreateABookmarkOfAUser", fmt.Sprintf("update a bookmark(%#v) occurs error", bookmark))
-	}
+	}*/
```
[the remark this answers is not in the document]
Comment (babygoat) [reply storage/bookmark.go]: @nickhsine Do we need this operation to update? I believe it duplicated of the operation firstOrCreate above. My intention is to remove this section.

[the remark this answers is not in the document]
Comment (nickhsine) [reply storage/bookmark.go]: This update is on purpose.
The reason behind this update is to fresh the bookmark fields, such as `title`, `desc` ... etc, that might change due to editing.
However, this update could be discussed.

[the remark this answers is not in the document]
Comment (babygoat) [reply storage/bookmark.go]: Where do we perform the bookmark update? I could not find the related update place in controller layer.

[the remark this answers is not in the document]
Comment (nickhsine) [reply storage/bookmark.go]: After discussion: remove these business logic.
TODO: find another way to update the outdated bookmark.

[the remark this answers is not in the document]
Comment (babygoat) [reply storage/bookmark.go]: Issue open in #351 .

**Pull request merged**